scripts/genCurrencies.mjs

```diff
@@ -20,54 +22,103 @@ import { getCurrencyData } from './getCurrencyData.mjs';
 // eslint-disable-next-line import/no-named-as-default-member
 const { format, resolveConfig } = prettier;
 
+const baseOutputDir = path.resolve(
+  url.fileURLToPath(new URL('.', import.meta.url)), // __dirname es module style
+  '../packages/currencies/src/iso4217/amendments/168'
+);
+
 const dirExists = (dirPath) =>
   fs
     .stat(dirPath)
     .then((stats) => stats.isDirectory())
     .catch(() => false);
 
+/** @type {import("./genCurrencies").CreateCurrencyConfig} */
+function createCurrencyConfig({ genericType, imports, mapNumber }) {
+  return {
+    genericType,
+    generateFn: ({ code, description, ...currency }) => `
+      import type { Currency } from '../../../../types';
+      ${imports ? imports.map((text) => `${text}\n`) : ''}
+      /**
+       * ${description}.
+       */
+      export const ${code}: Currency<${genericType}> = {
+        code: '${code}',
+        base: ${mapNumber(currency.base)},
+        exponent: ${mapNumber(currency.exponent)},
+      };`,
+  };
+}
+
+const numberCurrencyConfig = createCurrencyConfig({
+  genericType: 'number',
+  mapNumber: (n) => `${n}`,
+});
+
+const bigintCurrencyConfig = createCurrencyConfig({
+  genericType: 'bigint',
+  mapNumber: (n) => `${n}n`,
```

Comment:
@reckter I implemented it, I was just doing it to test, but I pushed it with everything else. Hopefully thats okay!

Reply:
That's very ok! :D